src/main/java/be/seeseemelk/mockbukkit/ServerMock.java

```diff
@@ -298,6 +298,18 @@ public void addPlayer(@NotNull PlayerMock player)
 		try
 		{
 			conditionLatch.await();
+			if (preLoginEvent.getLoginResult() != AsyncPlayerPreLoginEvent.Result.ALLOWED)
+			{
+				PlayerConnectionCloseEvent playerConnectionCloseEvent =
+						new PlayerConnectionCloseEvent(player.getUniqueId(),
+								player.getName(),
+								player.getAddress().getAddress(),
+								false);
+
+				getPluginManager().callEvent(playerConnectionCloseEvent);
```

Comment:
Should this be called async?

Reply:
Is there a reason why it shouldn't

Reply:
From documentation it says that the event may be both synchronous or asynchronous